docs/sdk/src/guides/enable_elastic_scaling.rs

```diff
@@ -0,0 +1,162 @@
+//!             RELAY_PARENT_OFFSET
+//!         }
+//!     }
+//! ```
+//!
+//! ### Block production configuration
+//!
+//! This configuration directly controls the minimum block time and maximum number of cores
+//! the parachain can use.
+//!
+//! Example configuration for a 3 core parachain:
+//!  ```rust
+//!     /// How many parachain blocks are processed by the relay chain per parent. Limits the
+//!     /// number of blocks authored per slot. This determines the minimum block time of the
+//!     /// parachain: `RELAY_CHAIN_SLOT_DURATION_MILLIS/BLOCK_PROCESSING_VELOCITY`
+//!     const BLOCK_PROCESSING_VELOCITY: u32 = 3;
+//!
+//!     /// Maximum number of blocks simultaneously accepted by the Runtime, not yet included
+//!     /// into the relay chain.
+//!     const UNINCLUDED_SEGMENT_CAPACITY: u32 = 2 * BLOCK_PROCESSING_VELOCITY + 1;
+//!
+//!     /// Relay chain slot duration, in milliseconds.
+//!     const RELAY_CHAIN_SLOT_DURATION_MILLIS: u32 = 6000;
+//!
+//!     type ConsensusHook = cumulus_pallet_aura_ext::FixedVelocityConsensusHook<
+//!         Runtime,
+//!         RELAY_CHAIN_SLOT_DURATION_MILLIS,
+//!         BLOCK_PROCESSING_VELOCITY,
+//!         UNINCLUDED_SEGMENT_CAPACITY,
+//!     >;
+//!
+//!  ```
+//!
+//! ## Current limitations
+//!
+//! ### Maximum execution time per relay chain block.
+
+//! Since parachain block authoring is sequential, the next block can only be built after
+//! the previous one has been imported.
+//! At present, a core allows up to 2 seconds of execution per relay chain block.
+//!
+//! If we assume a 6s parachain slot, and each block takes the full 2 seconds to execute,
+//! the parachain will not be able to fully utilize the compute resources of all 3 cores.
+//!    
+//! If the collator hardware is faster, it can author and import full blocks more quickly,
+//! making it possible to utilize even more than 3 cores efficiently.
+//!
+//! #### Why?
+//!
+//! Within a 6-second parachain slot, collators can author multiple parachain blocks.
+//! Before building the first block in a slot, the new block author must import the last
+//! block produced by the previous author.
+//! If the import of the last block is not completed before the next relay chain slot starts,
+//! the new author will build on its parent (assuming it was imported).
+//! This means that, on reference hardware, a parachain with a slot time of 6s can
+//! effectively utilize up to 4 seconds of execution per relay chain block, because it needs to
+//! ensure the next block author has enough time to import the last block.
+//! Hardware with higher single-core performance can enable a parachain to fully utilize more
+//! cores.
+//!
+//! ### Fixed factor scaling.
+//!
+//! For true elasticity, a parachain needs to acquire more cores when needed in an automated
+//! manner. This functionality is not yet available in the SDK, thus acquiring additional
+//! on-demand or bulk cores has to be managed externally.
```

Comment:
I'd also add that elastic scaling does not work with more than one on-demand core. Additional on-demand orders are always placed to the same core